e2e/internal/lb/lb_test.go

```diff
@@ -0,0 +1,159 @@
+//go:build e2elb
+
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/e2e/internal/kubectl"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	coreV1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	namespaceName = "lb-test"
+	serviceName   = "whoami"
+	initialPort   = int32(8080)
+	newPort       = int32(8044)
+	numRequests   = 256
+	numPods       = 3
+)
+
+func TestLoadBalancer(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	k, err := kubectl.New()
+	require.NoError(err)
+
+	// Wait for external IP to be registered
+	svc := testEventuallyExternalIPAvailable(t, k)
+	loadBalancerIP := svc.Status.LoadBalancer.Ingress[0].IP
+	loadBalancerPort := svc.Spec.Ports[0].Port
+	require.Equal(initialPort, loadBalancerPort)
+	url := buildURL(t, loadBalancerIP, loadBalancerPort)
+	testEventuallyStatusOK(t, url)
+
+	// Check that all pods receive traffic
+	var allHostnames []string
+	for i := 0; i < numRequests; i++ {
+		allHostnames = testEnpointAvailable(t, url, allHostnames)
+	}
+	assert.True(hasNUniqueStrings(allHostnames, numPods)) // check all pods receive traffic
+	allHostnames = allHostnames[:0]
```

Comment:
I think both are correct, but my syntax will simply "reset" the slice while keeping the underlying buffer. Since we store the same number of results from GET requests no additional allocations or copies are necessary.

Reply:
Might be more efficient, but makes reading harder...